src/arrays.js

```diff
@@ -1,61 +1,82 @@
 const getNthElement = (index, array) => {
-  // your code here
+  if (index >= array.length) 
+  return array[index - array.length];
+  return array[index]
 };
 
 const arrayToCSVString = array => {
-  // your code here
+  return array.toString()
 };
 
-const csvStringToArray = string => {
-  // your code here
-};
+function csvStringToArray(string) {
+  return string.split(",", 5)
+}
 
-const addToArray = (element, array) => {
-  // your code here
+const addToArray = (elementX, arrayX) => {
+  arrayX.push(elementX)
 };
 
-const addToArray2 = (element, array) => {
-  // your code here
+const addToArray2 = (NEWelement, ORIGarray) => {
+  let addToArray3;
+  addToArray3 = ORIGarray.concat(NEWelement)
+  return addToArray3
 };
 
 const removeNthElement = (index, array) => {
-  // your code here
+ return array.splice(index,1)
 };
 
+
 const numbersToStrings = numbers => {
-  // your code here
+  return numbers.map(String)
 };
 
 const uppercaseWordsInArray = strings => {
-  // your code here
+  return strings.map(function(A){return A.toUpperCase();})
 };
 
 const reverseWordsInArray = strings => {
-  // your code here
+   return strings.map(function(A){return A.split("").reverse().join("");})
 };
 
 const onlyEven = numbers => {
-  // your code here
+//const numbers2 = numbers.filter(a => a % 2 === 0); works, replaced filter with map,but this gives T or F//
+return numbers.filter(a => a % 2 === 0);
 };
 
+//splice() method mutates an array and returns the removed items. Splice example with no mutation.
 const removeNthElement2 = (index, array) => {
-  // your code here
+let newArray = [...array]
+if (index >= 0) {
+    newArray.splice(index, 1)
+    return newArray
+  }
 };
 
 const elementsStartingWithAVowel = strings => {
-  // your code here
+  let results = []
+  for (let i = 0; i <strings.length; i++) {
+    if ((strings[i].startsWith('a')) || (strings[i].startsWith('e')) || (strings[i].startsWith('i'))  
+    || (strings[i].startsWith('o')) || (strings[i].startsWith('u')) || (strings[i].startsWith('A')) 
+    || (strings[i].startsWith('E')) || (strings[i].startsWith('I')) || (strings[i].startsWith('O')) 
+    || (strings[i].startsWith('U')))
+    results.push(strings[i])
+    }
+    return results
 };
 
 const removeSpaces = string => {
-  // your code here
+  return string.replace(/\s+/g, ''); // Research this solution 
```

Comment:
This is regex, quite a big topic. This is one of those things that you'll end up using at some point, its a very versatile and powerful way of matching patterns in strings very quickly. They are written in a format like `/[aeiou]/`, and once you get the hang of them they are really useful

What your code here means is: find any whitespace (the `s` character, which has to be prefixed with the `\` otherwise it will match with the actual letter s) in the given string that occurs 1 or more times (this is the `+` characters job), and look for all matches in the string (this is the `g` flags job at the end)

Youll come across this more at a later date Im sure, dont worry about learning the ins and outs of this too much at this stage. Regardless, comment should be removed :)